### PR TITLE
vim-patch:7.4.1546, 7.4.1548

### PR DESCRIPTION
**Pull request by prollings**
vim-patch:7.4.1546

Problem:    Sticky type checking is more annoying than useful.
Solution:   Remove the error for changing a variable type.

https://github.com/vim/vim/commit/f6f32c38bf3319144a84a01a154c8c91939e7acf

Note: There are a bunch of other changes to eval.txt that I believe are
N/A and not related to this patch.

vim-patch:7.4.1548

Problem:    Two tests fail.
Solution:   Adjust the expected error number. Remove check for type.

https://github.com/vim/vim/commit/5a2800fd141a8fc0c80cdf421dcb76001a22327f

### CODE REVIEW

File: src/nvim/testdir/Makefile [hunk 1]
```diff
@@ -33,9 +33,11 @@ SCRIPTS := \
 # Tests using runtest.vim.vim.
 # Keep test_alot*.res as the last one, sort the others.
 NEW_TESTS = \
+	    test_assign.res \
 	    test_cursor_func.res \
 	    test_hardcopy.res \
 	    test_help_tagjump.res \
+	    test_history.res \
```
Comment (KillTheMule) [inline src/nvim/testdir/Makefile]: This is in your other PR :)

**Comment by KillTheMule** [comment]
Test integration looks fine to me

**Comment by KillTheMule** [comment]
The tests need to be adjusted, see https://github.com/vim/vim/commit/5a2800fd141a8fc0c80cdf421dcb76001a22327f. Maybe also https://github.com/vim/vim/commit/3905e291fe4375ca5c59efa9ffcb01a39c7be3a9.

File: src/nvim/testdir/Makefile (hunk 1 above)
Comment (prollings) [inline src/nvim/testdir/Makefile]: Haha, interesting... :P

**Comment by prollings** [comment]
@KillTheMule That makes sense. I'll package in those patches to this PR.

**Comment by KillTheMule** [comment]
In that case, you should relabel this PR as [WIP].

File: src/nvim/testdir/Makefile [hunk 2]
```diff
@@ -33,6 +33,7 @@ SCRIPTS := \
 # Tests using runtest.vim.vim.
 # Keep test_alot*.res as the last one, sort the others.
 NEW_TESTS = \
+	    test_assign.res \
```
Comment (justinmk) [inline src/nvim/testdir/Makefile]: This should be in `test_alot.vim`. I fixed it and others that were here.

**Pull request merged**